server/core/include/irods/configuration_hook_manager.hpp

```diff
@@ -0,0 +1,65 @@
+#ifndef CONFIGURATION_HOOK_MANAGER_HPP
+#define CONFIGURATION_HOOK_MANAGER_HPP
+
+#include <vector>
+#include <string>
+#include <nlohmann/json.hpp>
+/// \file
+
+namespace irods::experimental::hooks{
+  /**
+
+   */
+  using hook =
+    std::function<void(const std::string&,const nlohmann::json&,const nlohmann::json&)>;
+  ///
+  ///Provides a mechanism to trigger the execution of code
+  ///when a value in the configuration changes.
+  ///
+  ///Hooks are responsible for determining 
+  class configuration_hook_manager{
```

Comment:
Why use a singleton class?
Why not use free functions?

For example, see the following:
- https://github.com/irods/irods/blob/4.3.0/lib/core/include/irods/hostname_cache.hpp
- https://github.com/irods/irods/blob/4.3.0/server/core/include/irods/replica_access_table.hpp

Reply:
I always forget I can just have some variables tucked away in a namespace. :)